### PR TITLE
Feature/percy screenshots

### PR DESCRIPTION
**Pull request by dtjones404**
### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues
[DTPPSDK-990](https://paypal.atlassian.net/browse/DTPPSDK-990)

Percy screenshot integration has been added as a proof-of-concept.  At present, Percy screenshots will be captured in parallel with our existing screenshot test suite and will be non-blocking for PRs- this will allow us to evaluate Percy's functionality without impeding the work of other engineers.  It also prevents any blocking issues that might occur if we exceed our monthly screenshot limits through Browserstack.  Percy screenshots have also been disabled for PRs created by Renovate-bot in order to conserve monthly screenshots.
### Screenshots (if applicable)
Screenshot 2023-01-30 at 2 16 24 PM

### Groups who should review (if applicable)

❤️  Thank you!

### CODE REVIEW

File: test/percy/tests/screenshot.spec.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    scope: ".paypal-buttons",
+  });
+};
+
+const buttonConfigs = JSON.parse(
+  // eslint-disable-next-line no-sync
+  fs.readFileSync("./test/percy/files/buttonConfigs.json")
+);
+
+test.setTimeout("600000");
```
Comment (westeezy) [inline test/percy/tests/screenshot.spec.js:96]: Was this timeout needed?

Comment (dtjones404) [reply test/percy/tests/screenshot.spec.js:96]: I was seeing a bug where the test would close before all screenshots were uploaded to Percy and adding a timeout seems to have fixed it.  It probably doesn't need to be this long, though!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -44,6 +44,12 @@ jobs:
       - name: ▶️ Run Jest Screenshot
         run: npm run jest-screenshot
 
+      - name: ▶️ Run Percy Screenshot
+        if: ${{ github.actor != 'renovate[bot]' }}
```
Comment (gregjopa) [inline .github/workflows/main.yml:48]: Nice work figuring out how to stop renovate from using all of our budget for Percy screenshots 💯

File: test/percy/tests/screenshot.spec.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  }
+
+  await percySnapshot(page, description, {
+    scope: ".paypal-buttons",
+  });
+};
+
+const buttonConfigs = JSON.parse(
+  // eslint-disable-next-line no-sync
+  fs.readFileSync("./test/percy/files/buttonConfigs.json")
```
[the remark this answers is not in the document]
Comment (dtjones404) [reply test/percy/tests/screenshot.spec.js:92]: Seems to work great but testing on fork just to be sure

**Comment by gregjopa** [comment]
@dtjones404 this PR is looking awesome. One request, can you add a README.md file to the new percy folder that explains what contributors need to do to add a new screenshot?

File: test/percy/tests/screenshot.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (dtjones404) [reply test/percy/tests/screenshot.spec.js]: looks good!

**Review by gregjopa — APPROVED** [review]
💯

**Pull request merged**